lib/uint64.js

```diff
@@ -0,0 +1,274 @@
+'use strict';
+
+const UINT32_MAX = 0xffffffff;
+
+const charToNum = {
+  0: 0, 1: 1, 2: 2,
+  3: 3, 4: 4, 5: 5,
+  6: 6, 7: 7, 8: 8,
+  9: 9,
+  a: 10, A: 10,
+  b: 11, B: 11,
+  c: 12, C: 12,
+  d: 13, D: 13,
+  e: 14, E: 14,
+  f: 15, F: 15,
+};
+
+class Uint64 {
+  constructor(value) {
+    this.value = new Uint32Array(2);
+    if (value instanceof Uint64) {
+      this.value[0] = value.value[0];
+      this.value[1] = value.value[1];
+      return;
+    }
+    const numValue = Number(value);
+    if (Number.isNaN(numValue) || numValue <= 0) {
+      return;
+    }
+    if (numValue <= UINT32_MAX) {
+      this.value[0] = numValue;
+    } else if (numValue <= Number.MAX_SAFE_INTEGER) {
+      this.value[0] = numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(numValue / (UINT32_MAX + 1));
+    } else if (typeof value === 'string') {
+      let radix = 10;
+      if (value.startsWith('0x')) {
+        radix = 16;
+      } else if (value.startsWith('0o')) {
+        radix = 8;
+      } else if (value.startsWith('0b')) {
+        radix = 2;
+      }
+      const uintRadix = new Uint64(radix);
+      let res = new Uint64();
+      for (let i = 0; i < value.length; i++) {
+        const digit = charToNum[value[i]];
+        const uintDigit = new Uint64(digit);
+        res = Uint64.mult(res, uintRadix).add(uintDigit);
```

Comment:
I am not sure if we need this, there is also a need to perform multiple checks and conversions (that are already being done in the constructor) on the numbers before they can be used in all of the operations defined on `Uint64` numbers.

Reply:
That will be much better API IMO, when there is no need to wrap values (those can just be separate methods that support or work exclusively for numbers).
Also, if we don't throw on incorrect values we can just use `value >>> 0` to convert basically anything to uint32 =). I don't think any more checks are performed in the constructor besides handling values from `UINT32_MAX` to `MAX_SAFE_INTEGER` but that may be skipped I think.

Reply:
This won't make it possible to use number values bigger than `0xffffffff`.

Reply:
Yep, only for valid uint32 values.

Reply:
I wanted to say that wrapping allows using numbers bigger than `0xffffffff`.

Reply:
Well, yes, I proposed having separate method for cases where you don't need such wrapping and therefore can use plain numbers. This will also improve some of the internal implemenions by avoiding manual wrapping.